lib/_stream_writable.js

```diff
@@ -538,20 +536,21 @@ function endWritable(stream, state, cb) {
 function CorkedRequest(state) {
   this.next = null;
   this.entry = null;
+  this.finish = onCorkedFinish.bind(undefined, this, state);
+}
```

Comment:
All of this logic is needed because we have a very slow `bind`. In a year or so, we might be able to remove this whole `CorkedRequest`  thing. I am more concerned of `readable-stream` rather than core.